### ISSUE TITLE
Add option to opt out of extensions

### ISSUE THREAD

**Issue by chrmarti** [microsoft/vscode-remote-release#9177]
Tracking https://github.com/devcontainers/features/issues/386 here.

**Comment by chrmarti** [comment]
I suggest we use the `-` notation suggested in https://github.com/microsoft/vscode-remote-release/issues/415 to omit extensions from being installed. This fits in the existing devcontainer.json and devcontainer-feature.json customizations, e.g., omitting one while adding another extension would look like:
```
"customizations": {
	"vscode": {
		"extensions": {
			"-dbaeumer.vscode-eslint",
			"ms-python.python"
		}
	}
}
```

Additions / removals of extensions in the devcontainer.json should take precedence over those in the devcontainer-feature.jsons and these should both take precedence over those in the base image.

**Comment by chrmarti** [comment]
Available in Dev Containers 0.323.0-pre-release.

**Comment by BAAZKonqi** [comment]
Should this prevent optional dependencies from being loaded?
I'm trying to load the C# Dev Kit extension without the IntelliCode extension, but it is still being installed:

```
"extensions": [
        "-ms-dotnettools.vscodeintellicode-csharp",
        "ms-dotnettools.csdevkit"
      ]
```
      
It is possible to uninstall IntelliCode manually and the C# Dev Kit stays functional.
